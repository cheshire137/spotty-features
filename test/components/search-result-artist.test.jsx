import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import ArtistSearchResponse from '../fixtures/spotify/artist-search'

import SearchResultArtist from '../../src/components/search-result-artist.jsx'

const artist = ArtistSearchResponse.artists.items[0]
const image = artist.images[3].url

function props(chooseArtist, select) {
  return {
    image,
    name: artist.name,
    selected: false,
    chooseArtist,
    select
  }
}

describe('SearchResultArtist', () => {
  let component = null
  let artistChosen = false
  let artistSelected = false

  beforeEach(() => {
    const chooseArtist = () => {
      artistChosen = true
    }
    const select = () => {
      artistSelected = true
    }
    component = <SearchResultArtist {...props(chooseArtist, select)} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('displays artist name', () => {
    const title = shallow(component).find('.track-name')
    expect(title.text()).toBe('Tom Petty')
  })

  test('displays artist art', () => {
    const images = shallow(component).find('.track-image')
    expect(images.length).toBe(1)
    expect(images.at(0).props().src).toBe(image)
  })

  test('search result can be chosen', () => {
    const button = shallow(component).find('.search-result-button')
    expect(artistChosen).toBe(false)
    button.simulate('click')
    expect(artistChosen).toBe(true)
  })

  test('search result can be selected', () => {
    const button = shallow(component).find('li')
    expect(artistSelected).toBe(false)
    button.simulate('mouseover')
    expect(artistSelected).toBe(true)
  })
})
