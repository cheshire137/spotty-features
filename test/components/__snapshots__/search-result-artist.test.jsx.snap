exports[`SearchResultArtist matches snapshot 1`] = `
<li
  className=""
  onMouseOver={[Function]}>
  <button
    className="search-result-button"
    onClick={[Function]}
    type="button">
    <span
      className="columns">
      <span
        className="column track-image-column">
        <img
          className="track-image"
          src="https://i.scdn.co/image/35033e0288ffdc1b0a0958cc917a93d5a28b8b8b" />
      </span>
      <span
        className="column search-result-details">
        <span
          className="track-name">
          Tom Petty
        </span>
      </span>
    </span>
  </button>
</li>
`;
